include/swift/IDE/CompletionInstance.h

```diff
@@ -129,14 +130,23 @@ class CompletionInstance {
   /// In case of failure or cancellation, the callback receives the
   /// corresponding failed or cancelled result.
   ///
+  /// If \p CancellationFlag is not \c nullptr, code completion can be cancelled
+  /// by setting the flag to \c true.
+  /// IMPORTANT: If \p CancellationFlag is not \c nullptr, then completion might
+  /// be cancelled in the secon pass that's invoked inside \p Callback.
+  /// Therefore, \p Callback MUST check whether completion was cancelled before
+  /// interpreting the results, since invalid results may be returned in case
+  /// of cancellation.
+  ///
   /// NOTE: \p Args is only used for checking the equaity of the invocation.
   /// Since this function assumes that it is already normalized, exact the same
   /// arguments including their order is considered as the same invocation.
   void performOperation(
-      swift::CompilerInvocation &Invocation, llvm::ArrayRef<const char *> Args,
+      swift::CompilerInvocation &&Invocation, llvm::ArrayRef<const char *> Args,
```

Comment:
I'm not against this change. But is this necessary for this PR? If not, I wouldn't include it in this PR.

Reply:
No, it’s not. It’s just some cleanup I did while creating the PR. I removed the commit.